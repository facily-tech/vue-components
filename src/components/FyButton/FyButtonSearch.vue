<template>
  <fy-button
    outlined
    :color="color"
    class="fy-button-search"
    :data-test="dataTest"
    @click="$emit('click-filter')"
    v-bind="propsAndAttrs"
    v-on="$listeners"
  >
    <v-icon>mdi-magnify</v-icon>
    <span class="text-uppercase" :class="isHideName">
      {{ name }}
    </span>
  </fy-button>
</template>

<script lang="ts">
import Vue from 'vue';
import FyButton from './FyButton.vue';

export default Vue.extend({
  name: 'FyButtonSearch',

  components: {
    FyButton,
  },

  props: {
    name: {
      type: String,
      default: 'Buscar',
    },

    hideName: {
      type: Boolean,
      default: false,
    },

    dataTest: {
      type: String,
      default: 'btn-filter',
    },
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },

    isHideName() {
      return this.hideName ? 'd-none d-sm-flex' : '';
    },

    color(): string {
      return this.$vuetify?.theme?.dark ? 'grey' : 'primary';
    }
  },
});
</script>
