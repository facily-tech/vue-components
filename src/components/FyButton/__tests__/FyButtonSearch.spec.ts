import Vue from 'vue';
import Router from 'vue-router';
import FyButtonSearch from '../FyButtonSearch.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyButtonSearch', () => {
  // eslint-disable-next-line
  let router: Router;
  let localVue: typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);
  });

  it('should render component and match snapshot', () => {
    const wrapper = mount(FyButtonSearch, {
      localVue,
      vuetify: new Vuetify(),
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('when hideName should apply class', () => {
    const wrapper: Wrapper<any> = mount(FyButtonSearch, {
      localVue,
      vuetify: new Vuetify(),
      router,
      propsData: {
        hideName: true,
      },
    });

    expect(wrapper.vm.isHideName).toBe('d-none d-sm-flex');
  });

  it('when button is clicked', () => {
    const wrapper: Wrapper<any> = mount(FyButtonSearch, {
      localVue,
      vuetify: new Vuetify(),
      router,
    });

    expect(wrapper.emitted('click-filter')).toBeFalsy();

    wrapper.find('[data-test="btn-filter"]').trigger('click');

    expect(wrapper.emitted('click-filter')).toBeTruthy();
  });
});
