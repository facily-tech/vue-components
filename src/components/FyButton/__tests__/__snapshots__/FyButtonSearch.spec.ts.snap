// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyButtonSearch should render component and match snapshot 1`] = `
<button type="button" class="fy-button-search v-btn v-btn--outlined theme--light elevation-0 v-size--default primary--text" style="min-height: 40px;" data-test="btn-filter" name="Buscar" outlined="" color="primary"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-magnify theme--light"></i> <span class="text-uppercase">
    Buscar
  </span></span></button>
`;
